autoload/magit/utils.vim

```diff
@@ -20,12 +20,43 @@ function! magit#utils#is_submodule(dirname)
 	return ( index(s:submodule_list, a:dirname) != -1 )
 endfunction
 
-" s:magit_cd_cmd: plugin variable to choose lcd/cd command, 'lcd' if exists,
-" 'cd' otherwise
-let s:magit_cd_cmd = exists('*haslocaldir') && haslocaldir() ? 'lcd ' : 'cd '
-" magit#utils#lcd: helper function to lcd. use cd if lcd doesn't exists
-function! magit#utils#lcd(dir)
-	execute s:magit_cd_cmd . a:dir
+" magit#utils#chdir will change the directory respecting
+" local/tab-local/global directory settings.
+function! magit#utils#chdir(dir)
+  " This is a dirty hack to fix tcd breakages on neovim.
+  " Future work should be based on nvim API and has a `if has('nvim')` check.
+  if has('nvim')
+    let chdir = haslocaldir() ? 'lcd' : haslocaldir(-1, 0) ? 'tcd' : 'cd'
+  else
+    let chdir = exists('*haslocaldir') && haslocaldir() ? 'lcd' : 'cd'
```

Comment:
Well, I won't fight against this, in the end this is only a micro optim.

One day, I hope changing directory won't be necessary... I already tried to play around with GIT_WORK_TREE and GIT_DIR environment variables, but it was a dead end, I can't remember why.
I should give another try: in case it works, vimagit won't need to change dir anymore.